### ISSUE TITLE
Codefix for suppress <ID> in Source inserts pragma restore before semicolon

### ISSUE THREAD

**Issue by JohanLarsson** [dotnet/roslyn#26015]
Code before:
```cs
namespace RoslynSandbox
{
    public class Foo
    {
        public Foo()
        {
        var text = @"Line 1
Line 2";
        }
    }
}
```

Fixed code:
```cs
namespace RoslynSandbox
{
    public class Foo
    {
        public Foo()
        {
#pragma warning disable CS0219 // Variable is assigned but its value is never used
            var text = @"Line 1
Line 2"
#pragma warning restore CS0219 // Variable is assigned but its value is never used
;
        }
    }
}
```

I would prefer
Fixed code:
```cs
namespace RoslynSandbox
{
    public class Foo
    {
        public Foo()
        {
#pragma warning disable CS0219 // Variable is assigned but its value is never used
            var text = @"Line 1
Line 2";
#pragma warning restore CS0219 // Variable is assigned but its value is never used
        }
    }
}
```

The example uses the code fix to suppress `CS0219`

**Comment by CyrusNajmabadi** [comment]
The system we have for inserting pragmas should probably be updated to recognize statements and to know to not split the statement from its trailing semicolon.

**Comment by jcouv** [comment]
@JohanLarsson Thanks for reporting.
The fix was merged for 15.8.
